src/OmniSharp.Roslyn.CSharp/Services/CreateNewType/CreateNewTypeService.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+using System.Composition;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.Extensions.Logging;
+using OmniSharp.Mef;
+using OmniSharp.Models.CreateNewTypeRequest;
+
+namespace OmniSharp.Roslyn.CSharp.Services.CreateNewType
+{
+    [OmniSharpHandler(OmniSharpEndpoints.CreateNewType, LanguageNames.CSharp)]
+    public class CreateNewTypeService : IRequestHandler<CreateNewTypeRequest, CreateNewTypeResponse>
+    {
+        private const string FILE_CONTENT_TEMPLATE =
+@"namespace {0};
```

Comment:
It requires C#10 version and it's better to have template version for old lang version. But I can not find a way to get it from project.